engines/python/setup/djl_python/deepspeed.py

```diff
@@ -236,17 +300,71 @@ def infer_task_from_model_architecture(self, config: PretrainedConfig):
                 f"Task could not be inferred from model config. "
                 f"Please manually set `task` in serving.properties.")
 
+    def get_model_pretrained(self, model_id_or_path, torch_dtype='auto', **kwargs):
+        tokenizer = AutoTokenizer.from_pretrained(model_id_or_path)
+        model = TASK_TO_MODEL[self.task].from_pretrained(model_id_or_path, torch_dtype=torch_dtype, **kwargs)
+        return model, tokenizer
+
+    def get_model_from_config(self, model_id_or_path, **kwargs):
+        tokenizer = AutoTokenizer.from_pretrained(model_id_or_path)
+        model = TASK_TO_MODEL[self.task].from_config(self.model_config, **kwargs)
+        return model, tokenizer
+
+    def get_model(self, model_id_or_path, loading_method, **kwargs):
+        if loading_method == 'from_config':
+            return self.get_model_from_config(model_id_or_path, **kwargs)
+        elif loading_method == 'pretrained':
+            return self.get_model_from_config(model_id_or_path, **kwargs)
+        else:
+            raise RuntimeError(f'Unsupported model loading method, this should not happen.')
+
+    def load_model(self, model_id_or_path, loading_method, use_mmap_loader, **kwargs):
+        def load_model_with_mmap(model_id_or_path, loading_method):
+            import mmaploader
+            import accelerate
+            with mmaploader.load_mmap_meta() as mmap_loader:
+                with accelerate.init_empty_weights():
+                    kwargs['low_cpu_mem_usage'] = False
+                    model, tokenizer = self.get_model(model_id_or_path, loading_method, **kwargs)
+            return model, tokenizer, mmap_loader.state_dict_mmap
+
+        state_dict_mmap = {}
+        model = None
+        tokenizer = None
+        done = False
+
+        if use_mmap_loader:
+            try:
+                model, tokenizer, state_dict_mmap = load_model_with_mmap(model_id_or_path, loading_method)
+                done = True
+            except:
+                self.logger.warning(f'failed to load model with mmap loader, will load model normally')
```

Comment:
When do we enter here? What if the loading get into unrecoverable states?

Reply:
The loading fails because:
* Not all model formats are supported. We could limit it to ValueError or something if you'd prefer to cover this case at least.
* mmaploader is not available or fails to import for some reason.

Reply:
If the state is "unrecoverable", how do you think the behavior would be better than what will happen if we just try falling back to normal model loading?